importers/djornl/parser.py

```diff
@@ -164,19 +174,46 @@ def check_headers(self, headers, validator=None):
 
         :param validator: (obj)         validator object, with the appropriate schema loaded
 
-        :return missing_headers: (list) list of required headers that are missing from the input.
+        :return header_errs: (dict)     dict of header errors:
+                                        'missing': required headers that are missing from the input
+                                        'invalid': additional headers that should not be in the input
+                                        'duplicate': duplicated headers (content would be overwritten)
                                         If the list of headers supplied is valid--i.e. it
                                         contains all the fields marked as required in the validator
                                         schema--or no validator has been supplied, the method
-                                        returns an empty list
+                                        returns an empty dict
         """
 
         if validator is None:
-            return []
+            return {}
+
+        header_errs = {}
+
+        all_headers = {}
+        # ensure we don't have any duplicate headers
+        for h in headers:
+            if h in all_headers:
+                all_headers[h] += 1
+            else:
+                all_headers[h] = 1
+
+        duplicate_headers = [h for h in all_headers.keys() if all_headers[h] != 1]
+        if duplicate_headers:
+            header_errs['duplicate'] = duplicate_headers
 
         # check that each required header in the schema is present in headers
         required_props = validator.schema['required']
-        return [i for i in required_props if i not in headers]
+        missing_headers = [i for i in required_props if i not in headers]
+        if missing_headers:
+            header_errs['missing'] = missing_headers
+
+        if 'additionalProperties' in validator.schema and validator.schema['additionalProperties'] is False:
+            all_props = validator.schema['properties'].keys()
```

Comment:
unless `additionalProperties` is present and set to `False`, there is no such thing as an invalid header